### PR TITLE
Add a base64 decoder to strings processor

### LINKED ISSUE #4697: Add base64 decoder processor
## Feature Request
Add a base64 (or configurable) decoder processor

### Proposal:
Add to strings (not likely the correct place) a decoder, to decode a field.

### Example config:
```toml
[agent]
  interval="1s"
  flush_interval="1s"

[[inputs.exec]]
  timeout = "1s"
  data_format = "json"
  commands = [
    "echo -en '{\"message\":{\"attributes\":{\"deviceId\":\"myPi\",\"deviceNumId\":\"2808946627307959\",\"deviceRegistryId\":\"my-registry\",\"deviceRegistryLocation\":\"us-central1\",\"projectId\":\"conference-demos\",\"subFolder\":\"\"},\"data\":\"dGVzdGluZ0dvb2dsZSxzZW5zb3I9Ym1lXzI4MCB0ZW1wX2M9MjMuOTUsaHVtaWRpdHk9NjIuODMgMTUzNjk1Mjk3NDU1MzUxMDIzMQ==\",\"messageId\":\"204004313210337\",\"message_id\":\"204004313210337\",\"publishTime\":\"2018-09-14T19:22:54.587Z\",\"publish_time\":\"2018-09-14T19:22:54.587Z\"},\"subscription\":\"projects/conference-demos/subscriptions/my-subscription\"}'"
  ]
  tag_keys = ["message_attributes_deviceId"]
  json_string_fields = ["message_data"]

[[processors.strings]]
  [[processors.strings.decode]]
    field = "message_data"
    dest  = "message_data"

[[processors.parser]]
  parse_fields = ["message_data"]
  # change to "json" if the encoded data contains json
  data_format="influx"
  merge="override"

[[outputs.file]]
  files = ["stdout"]

## Sample output
# testingGoogle,host=test,message_attributes_deviceId=myPi,sensor=bme_280 message_data="testingGoogle,sensor=bme_280 temp_c=23.95,humidity=62.83 1536952974553510231",temp_c=23.95,humidity=62.83 1536972180000000000
```

### Desired behavior:
Decode a field so further processing can be done on it.

### Use case:
Google IOT sends a data field base64 encoded, it would be nice to be able to pull this data out and store it.

### PR DESCRIPTION
**Pull request by glinton**
Resolves #4697  

Add a base64 decoder processor to the strings processor plugin. Giving the ability to process data contained in a base64 encoded field. (Useful for working with google core iot)

### CODE REVIEW

**Comment by davidgs** [comment]
Brilliant! Thanks!

File: plugins/processors/strings/strings.go [hunk 1]
```diff
@@ -16,6 +17,7 @@ type Strings struct {
 	TrimRight  []converter `toml:"trim_right"`
 	TrimPrefix []converter `toml:"trim_prefix"`
 	TrimSuffix []converter `toml:"trim_suffix"`
+	Decode     []converter `toml:"decode"`
```
Comment (danielnelson) [inline plugins/processors/strings/strings.go]: Let's call it b64decode?

**Comment by danielnelson** [comment]
We can reopen this if it still is needed.

**Comment by cliffordvandyk** [comment]
It seems to be common practice (I have other examples than the original Google Core IOT) to send multiple IOT numeric payload fields combined into a single base64-encoded string. It would be useful to have methods to decode a base64 string, and then split the resulting (HEX) into multiple (InfluxDB-friendly) fields. Is there a reason why this feature request was apparently rejected? Can it be reopened?

**Comment by cliffordvandyk** [comment]
Also:
https://community.influxdata.com/t/telegraf-influxdb-parse-base-64-data-payload/2610

**Comment by danielnelson** [comment]
We could finalize this, can you comment over on the issue #4697

**Comment by cliffordvandyk** [comment]
Thanks, will do.

**Pull request closed without merging**